### PR TITLE
Define new project management workflow

### PR DESCRIPTION
**Pull request by tedsuo**
This pull request updates our current project management workflow, based on the proposal in #1534.

### CODE REVIEW

File: project-management.md
Comment (jpkrohling) [inline project-management.md:1]: Is this only for spec changes?

File: project-management.md
Comment (jpkrohling) [inline project-management.md:13]: Updated by whom? How frequent? What happens when it's not done?

File: project-management.md
Comment (jpkrohling) [inline project-management.md:33]: So, a project is a working group?

File: project-template.md
Comment (jpkrohling) [inline project-template.md:19]:
```suggestion
Projects cannot be started until the following participants have been identified:
```

**Review by jpkrohling — COMMENTED** [review]
I've reviewed the files at the root directory, not the individual projects.

File: project-management.md
Comment (tedsuo) [reply project-management.md:33]: A project is a working group, a goal, and a deadline. I can define this further.

**Comment by tedsuo** [comment]
@jpkrohling I rewrote the intro and Project proposal sections to be clearer about what counts as a project. Please have a look.

File: project-management.md
Comment (jpkrohling) [inline project-management.md:17]:
```suggestion
* Two TC/GC members, or community members delegated by them, to sponsor the project.
```

**Review by jpkrohling — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]
I like codifying this more explicitly.  We plan to leverage it more and more in semconv.

File: project-management.md
Comment (trask) [inline project-management.md:3]:
```suggestion
The OpenTelemetry community has limited bandwidth for managing changes which expand the scope of OpenTelemetry, or impact many SIGs within OpenTelemetry. There are three common scenarios which have this kind of impact.
```

File: project-template.md
Comment (trask) [inline project-template.md:21]:
```suggestion
* At least two sponsoring TC members. TC sponsors are dedicated to attending meetings, reviewing proposals, and in general being aware of the state of the project and its technical details. TC sponsors guide the project through the spec process, keep the tracking issue up to date, and help to ensure that relevant community members provide input at the appropriate times.
```

File: projects/completed-projects/http.md
Comment (trask) [inline projects/completed-projects/http.md:1]: This should probably be under projects/ instead of completed-projects/, we still need to deliver on Stage 3, and move from frozen to stable

**Review by trask — APPROVED** [review]
👍

File: projects/completed-projects/http.md
Comment (tedsuo) [reply projects/completed-projects/http.md:1]: no problem, I'll move it

**Comment by tigrannajaryan** [comment]
What is the process for existing projects/workgroups that did not follow this process?

For example we have Agent Management Workgroup that made great progress and has a lot of current momentum. I assume this proposal does not mean existing de-facto projects have to necessarily meet all the new requirements and have to pass a review before we accept them. We probably need to grandfather existing de-facto projects.

File: projects/client-instrumentation.md
Comment (MSNev) [inline projects/client-instrumentation.md:49]: The Wednesday meeting is not Monday @ 12:30am PST (for 30min) with Josh Sureth and Jack Berg

File: project-management.md
Comment (tedsuo) [reply project-management.md:15]: Deadlines do sound strict, but I think it will help the working groups focus and move towards delivering on their goals faster. Some things I have noticed when organizing prior projects:

* Having clear deadlines makes it easier for project members to receive approval to work on the project. When the work is open-ended, it is harder to get an approval from managers to use work time. This leads to engineers getting overburdened or pulled off the project. This has been particularly true for subject matter experts, who are participating in a particular project but are not planning on being a long term contributor to OpenTelemetry.
* Without a deadline, projects can become prone to "architectural astronauting," where they endlessly debate alternatives. Deadlines provide project leaders with a basis for ending debate and getting the group to produce a public proposal.
* Deadlines allow participants in the public review process to manage their calendars. Proper review can actually take a lot of time on the part of reviewers. Without deadlines, the public review period comes as a surprise, which leads to drawn out process due to critical participants being unavailable and unable to coordinate a response. This can be very frustrating to the project, as they are blocked on continuing their work until TC members and other approvers complete their review.

Obviously, a project can misjudge the amount of time they need, and can run into unexpected problems. In those cases, there is nothing wrong with updating the deadlines to something more realistic. But I think it is important that projects always maintain a clear expectation for when they will be delivering work for public review.

Comment (tigrannajaryan) [reply project-management.md:15]: Sounds good, thanks.

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
> > What is the process for existing projects/workgroups that did not follow this process?
> 
> @tigrannajaryan I propose that all existing projects fill out a project document and get added to the project board. I agree that there should not be an approval process - we have already agreed to do the work and have started the working group. But we need to surface all of these commitments so that we can track how thin our resources are spread.
> 
> Groups which currently do not have any deadlines would be encouraged to define them for any work which will require public review. The client and messaging groups found this helpful - having a six week goal led to an increased cadence in project work, and helped resolve debate. It will also help with getting a more coherent public review. If the deadlines prove to be unrealistic, they can be always be updated.

@tedsuo it would be useful to have this written in project-management.md so that the expectations from existing projects are clear.

File: projects/completed-projects/http.md
Comment (arminru) [reply projects/completed-projects/http.md:1]: ping @tedsuo 🙂

File: project-management.md
Comment (arminru) [inline project-management.md:6]:
```suggestion
* An existing SIG taking on new work which will affect OpenTelemetry as a whole, and will need review from the broader community.
```
or
```suggestion
* An existing SIG taking on new work which will affect the OpenTelemetry project as a whole, and will need review from the broader community.
```

File: project-management.md
Comment (arminru) [inline project-management.md:15]: Should these be part of the initial proposal or can these be laid out once the group has formed and work is starting?

File: project-management.md
Comment (arminru) [inline project-management.md:16]:
```suggestion
* A group of designers and subject matter experts need to dedicate a significant amount of their work time to the project. These participants are needed to design the spec, write a set of OTEPs, and create multiple prototypes. This group needs to meet with each other (and with their sponsors, see next point) on a regular basis to develop a successful set of proposals.
```

File: project-management.md
Comment (arminru) [inline project-management.md:27]: For this we would still need to create a ~dummy~ tracking issue in parallel to the PR adding the .md file, right?

File: project-template.md
Comment (arminru) [inline project-template.md:1]:
```suggestion
# <<PROJECT NAME>>
```

File: project-management.md
Comment (tedsuo) [reply project-management.md:27]: No, you don't need a tracking issue, you can just have a task directly in the project board.

**Pull request merged**
